identity/src/keypair.rs

```diff
@@ -724,4 +747,77 @@ mod tests {
 
         assert_implements_ord::<PublicKey>();
     }
+
+    #[test]
+    #[cfg(feature = "ed25519")]
+    fn test_publickey_from_ed25519_public_key_roundtrip_bytes() {
```

Comment:
I think we already have tests that go from and to bytes for public keys, do we not?

I'd like to keep these focused on one thing, i.e. the `From` and `Into` conversions in this case.

Reply:
There is a test above for protobuf bytes round trip, but not raw bytes (non-protobuf).

This is the whole reason I discovered the missing `From` issue was I was trying to get a `keypair::PublicKey` from raw bytes of a `ed25519::PublicKey`. This test uses `try_from_bytes` to generate that `ed25519::PublicKey` to test the `From`. 

I could have used keygen to generate the `ed25519::PublicKey`, but this test matches what we need in my real life scenario so I thought it was appropriate

Reply:
We don't really support exposing "raw bytes" as a first-class feature. libp2p applications should always export keys using the protobuf, otherwise the bytes are not "tagged" and thus it is unclear what they are.

Those functions primarily exist to allow importing keys from somewhere else.

Regardless of this, I think this is better done as two tests:

- One for the `From`
- One for creating a key from bytes

The latter is already tested in many other applications so I think we can just focus on the conversions here.

Thoughts?